### PR TITLE
Skip mode tests with BORG_TESTS_IGNORE_MODES

### LINKED ISSUE #1363: Tests failing on ntfs-3g
- [ ] No ACLs: `KeyError: 'acl_access'` pops up a bunch - also present on SSHFS (see #1362, fixed in #1373)
- [ ] Funky permissions and modes: (fixed in #1383)
  ```
                  # i have seen cases where the copied tree has world-readable
                  # permissions, which is wrong
>               assert stat_segment(backup).st_mode & UMASK_DEFAULT == 0
E               assert (33279 & 63) == 0

.tox/py35/lib/python3.5/site-packages/borg/testsuite/upgrader.py:180: AssertionError
  ```

  ```
          # Directory replaced with a regular file
>       assert '[drwxr-xr-x -> -rwxr-xr-x] input/dir_replaced_with_file' in output
E       AssertionError: assert '[drwxr-xr-x -> -rwxr-xr-x] input/dir_replaced_with_file' in '    +13 B       0 B input/empty\n    +13 B       0 B input/hardlink_contents_changed\n     +1 B
 -1.0 kB input/file_...moved       512 B input/file_removed2\nremoved directory   input/dir_removed\nremoved link        input/link_removed\n'
  ```

  ```
          mode = os.stat(self.repository_path).st_mode
>       self.assertEqual(stat.S_IMODE(mode), 0o700)
E       AssertionError: 0o777 != 0o700

.tox/py35/lib/python3.5/site-packages/borg/testsuite/archiver.py:995: AssertionError
```

- [x] No atime - fixed with #1367:

These should be fixed before #1359 is merged.

### PR DESCRIPTION
**Pull request by PlasmaPower**
Along with #1373, resolves #1363

If you cherry pick this and #1373 onto #1359, all tests pass! :tada:

### CODE REVIEW

**Comment by enkore** [comment]
lgtm

File: src/borg/testsuite/archiver.py [hunk 1]
```diff
@@ -988,6 +988,7 @@ def test_readonly_repository(self):
             # Restore permissions so shutil.rmtree is able to delete it
             os.system('chmod -R u+w ' + self.repository_path)
 
+    @pytest.mark.skipif('BORG_TESTS_UNRELIABLE_MODES' in os.environ, reason='modes unreliable')
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: that env var name is a bit confusing.
isn't it rather BORG_TESTS_SKIP_UNRELIABLE_MODES?
"borg tests unreliable modes" sounds rather like the opposite.

Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: also, at least a short comment in the source about why we have this would be useful for the future reader.

**Comment by ThomasWaldmann** [comment]
one minor thing, other than that it lgtm.

File: src/borg/testsuite/archiver.py (hunk 1 above)
Comment (PlasmaPower) [inline src/borg/testsuite/archiver.py]: I'm trying to indicate that the modes themselves are unreliable, not the tests on them. Maybe `BORG_TESTS_IGNORE_MODES`?

Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: yes, that is less confusing.

Comment (PlasmaPower) [inline src/borg/testsuite/archiver.py]: Fixed.

**Comment by PlasmaPower** [comment]
I'm going to merge this into #1373 so I can include this in the py.test header. This patch probably belongs there anyway.

**Pull request closed without merging**